main.go

```diff
@@ -130,21 +173,27 @@ func (p Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request, next caddyhttp.
 		return nil
 	}
 
-	to, err := p.authRedirect(r)
+	upstream, err := p.authRedirect(r)
 	if err != nil {
 		return err
 	}
-	caddyhttp.SetVar(r.Context(), "upstream", to)
-	p.log.Info("setting upstream to " + to)
+
+	if upstream == p.api || upstream.Path == "" {
+		upstream.Path += r.RequestURI
+	}
+
+	caddyhttp.SetVar(r.Context(), "upstream", upstream.To)
+	caddyhttp.SetVar(r.Context(), "upstream_path", upstream.Path)
+	p.log.Info("setting upstream to " + upstream.To + ", path to " + upstream.Path)
 
 	return next.ServeHTTP(w, r)
 }
 
-func (p Proxy) authRedirect(r *http.Request) (string, error) {
+func (p Proxy) authRedirect(r *http.Request) (AuthSite, error) {
 	// if no cookie, redirect to get new cookie
 	cookie, err := r.Cookie(p.auth.CookieName)
 	if err != nil {
-		return p.ManagementAPI, nil
+		return p.api, nil
```

Comment:
Out of curiosity, why the change from "ManagementAPI" to "api"?

Reply:
mainly needed so that I could get around the reverse_proxy and paths. It's quite possible I can remove this if/when we switch to redirect instead of reverse proxy to management api.